### PR TITLE
[datadog_integration_gcp] Support enabling the CSPM integration

### LINKED ISSUE #1743: Support CSPM resource collection in datadog_integration_gcp
Just like [datadog_integration_aws](https://registry.terraform.io/providers/DataDog/datadog/latest/docs/resources/integration_aws), datadog_integration_gcp should have a field to enable CSPM resource collection. Currently it's only in the UI

image

### PR DESCRIPTION
**Pull request by christophetd**
Context: While the API does support it, it's currently not possible to enable the GCP CSPM through Terraform. The AWS integration resource does have support for it: https://registry.terraform.io/providers/DataDog/datadog/latest/docs/resources/integration_aws#cspm_resource_collection_enabled

### CODE REVIEW

File: go.mod
Comment (christophetd) [inline go.mod:4]: Note: https://github.com/DataDog/datadog-api-client-go/pull/1843 has just been merged. I will update this to point to the latest version of datadog-api-client-go once a new release is published next week

**Comment by christophetd** [comment]
(closes #1743)

**Comment by christophetd** [comment]
👋🏼 Eager to get this PR reviewed, and will get it merged once the next release of the Go SDK lands next week

**Review by skarimo — DISMISSED** [review]

**Review by therve — DISMISSED** [review]

**Comment by christophetd** [comment]
Thanks for the review! Will wait for the Go SDK release next week, update the go.mod and then merge it

**Review by rtrieu — APPROVED** [review]

**Review by therve — APPROVED** [review]

**Pull request merged**
